lib/_http_server.js

```diff
@@ -475,7 +475,11 @@ function socketOnError(e) {
 function onParserExecuteCommon(server, socket, parser, state, ret, d) {
   resetSocketTimeout(server, socket, state);
 
+  if (!d)
+    d = parser.getCurrentBuffer();
```

Comment:
Can you move this into the branches of the if statement?  `parser.getCurrrentBuffer()` creates a copy.  Calling it when the result is unused is wasteful and probably has a pretty big performance impact.

Reply:
But if we don't get the buffer, we cannot pass it to the `clientError` event.